Add options to enable/disable docs and demos

Originally written by Anthony Viallard
<viallard@syscom-instruments.com>.

[Vincent: adapt patch to 5.0.5 version]

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>

diff -rup a/configure.in b/configure.in
--- a/configure.in	2016-09-07 05:14:29.000000000 +0100
+++ b/configure.in	2016-12-01 12:09:31.487464774 +0000
@@ -818,6 +818,16 @@ if test "$with_cwdrc" = yes; then
                [ Define if you want to read .gnuplot from current directory (SECURITY RISK!).])
 fi
 
+dnl Whether we want to build the doc files
+dnl Does nothing here, is passed on to the doc subdir
+AC_ARG_ENABLE(doc,dnl
+[  --disable-doc    do not build doc files])
+
+dnl Whether we want to build the demo files
+dnl Does nothing here, is passed on to the demo subdir
+AC_ARG_ENABLE(demo,dnl
+[  --disable-demo    do not build demo files])
+
 dnl Sort help/subtopic tables by row or column
 AC_ARG_WITH(row-help,dnl
 [  --with-row-help         format help and subtopic tables by row (default)
@@ -1303,6 +1313,24 @@ fi
 dnl No configuration option for this one yet
 AC_DEFINE(MAX_PARALLEL_AXES,7, [Maximum number of parallel axes supported])
 
+dnl build doc files
+if test "$enable_doc" != no; then
+  AC_CONFIG_SUBDIRS(docs)
+  DOCSUBDIR=docs
+else
+  DOCSUBDIR=
+fi
+AC_SUBST(DOCSUBDIR)
+
+dnl build demo files
+if test "$enable_demo" != no; then
+  AC_CONFIG_SUBDIRS(demo)
+  DEMOSUBDIR=demo
+else
+  DEMOSUBDIR=
+fi
+AC_SUBST(DEMOSUBDIR)
+
 dnl Substitute variables
 AC_SUBST(PACKAGE)
 AC_SUBST(VERSION_MAJOR)
diff -rup a/Makefile.am b/Makefile.am
--- a/Makefile.am	2014-05-30 17:38:08.000000000 +0100
+++ b/Makefile.am	2016-12-01 12:09:31.488464794 +0000
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in -*-Makefile-*-
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = config m4 term src docs $(LISPDIR) man demo tutorial share
+SUBDIRS = config m4 term src $(DOCSUBDIR) $(LISPDIR) man $(DEMOSUBDIR) tutorial share
 
 EXTRA_DIST = BUGS CodeStyle Copyright FAQ.pdf GNUmakefile INSTALL INSTALL.gnu \
 Makefile.maint PATCHLEVEL PGPKEYS PORTING README RELEASE_NOTES \
